src/models/models.series.js

```diff
@@ -93,6 +93,17 @@ export default class ModelsSeries extends ModelsBase {
     }
 
     if (this._seriesInstanceUID === series.seriesInstanceUID) {
+      // may merge incorrectly if loader will return more than one stacks per series
+      if (this._stack[0] && series.stack[0]) {
```

Comment:
Do we have to do it on `this._stack` and not only the stack we want to merge into `this._stack`?